ironic/drivers/modules/drac/management.py

```diff
@@ -387,5 +400,77 @@ def known_good_state(self, task):
         node = task.node
 
         client = drac_common.get_drac_client(node)
+        if client.is_lifecycle_in_recovery():
+            LOG.info("Lifecycle Controller is in recovery mode")
+            get_lcc_out_of_recovery_mode(node)
         client.reset_idrac(force=True, wait=True)
         client.delete_jobs(job_ids=[_CLEAR_JOB_IDS])
+
+    @METRICS.timer('DracManagement.get_lifecycle_out_of_recovery')
+    @base.clean_step(priority=0)
+    def get_lifecycle_out_of_recovery(self, task):
```

Comment:
Would you mind changing "get_lifecycle_out_of_recovery" to "take_lifecycle_out_of_recovery"?

Reply:
Sure, I will change the method name.